hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/states/ContainerEntry.java

```diff
@@ -31,7 +31,8 @@
  */
 public class ContainerEntry {
   private final ContainerInfo info;
-  private final Map<DatanodeID, ContainerReplica> replicas = new HashMap<>();
+  private Map<ContainerReplica, ContainerReplica> replicas = new HashMap<>();
```

Comment:
`replicas` can be created lazily by the first modification, until then an empty map is fine.

```suggestion
  private Map<ContainerReplica, ContainerReplica> replicas = Collections.emptyMap();
```